### ISSUE TITLE
Souffle 1.6.0 stuck on RAM transformer sequence

### ISSUE THREAD

**Issue by gfour** [souffle-lang/souffle#1093]
Souffle 1.6.0 cannot compile the attached Doop analysis ([gen_5073977971029498271.dl.zip](https://github.com/souffle-lang/souffle/files/3441242/gen_5073977971029498271.dl.zip)):
```
souffle -c -o exe gen_5073977971029498271
```

Analysis compilation times:
Souffle 1.5.1: 112 sec
Souffle 1.6.0: does not terminate after 9000 sec

Commenting out the following transformation sequence fixes the issue (I don't know how that affects program correctness):

```
std::unique_ptr<RamTransformer> ramTransform = std::make_unique<RamTransformerSequence>(
    ...
    // std::make_unique<RamLoopTransformer>(std::make_unique<RamTransformerSequence>(
    //   std::make_unique<HoistAggregateTransformer>(), std::make_unique<TupleIdTransformer>())),
    ...
```

**Comment by b-scholz** [comment]
We will check; for the time being you can unroll the transformer loop twice and you should have most performance opportunities resolved.

**Comment by b-scholz** [comment]
Note the correctness is not affected - only performance opportunities are not materalized.

**Comment by b-scholz** [comment]
The HoistAggregateTransformer pretends to hoist an aggregate from an inner loop although it cannot hoist it anymore. This rewrites the code and causes the endless loop.

**Comment by b-scholz** [comment]
We have found a small example that exposes the issue:
```
.decl A(x:number)
A(1).
A(2).

.decl B(x:number)
B(y0) :- 
   A( y0),  
   y0 < count : A(_).
.printsize B
```

The aggregate is hoisted - however, the old aggregate still remains in the RAM program.

**Comment by b-scholz** [comment]
There is another example related to the code:
```
.decl InvocationToInline,TripleInvocationTarget,Instruction_Method(x:number, y:number) input output
.decl CallerSize(x:number) input output

InvocationToInline(?invocation,?target) :-
      TripleInvocationTarget(?invocation,?target),
      Instruction_Method(?invocation,?invocationMethod),
      CallerSize(?invocationMethod),
      3 = count : TripleInvocationTarget(?invocation,_).
```

**Comment by b-scholz** [comment]
The second example related to the code:
```
.decl InvocationToInline,DoubleInvocationTarget,Instruction_Method(x:number, y:number) input output
.decl CallerSize(x:number) input output

InvocationToInline(?invocation,?target) :- 
     DoubleInvocationTarget(?invocation,?target),
     Instruction_Method(?invocation,?invocationMethod),
     CallerSize(?invocationMethod),
     2 = count : DoubleInvocationTarget(?invocation,_).
```

**Comment by b-scholz** [comment]
I made a fix - PR #1098. I still need to add the test-cases.

**Comment by b-scholz** [comment]
This should be fixed now.
